### PR TITLE
[V4] Prebuild the admin panel.

### PR DESCRIPTION
**Pull request by soupette**
### What does it do?

It fixes the build script, now only the plugins that are shipped with a generated app will be built.
It also introduces a script that writes the plugins.js  file before starting the webpack-dev-server when developing in the monorepo.

### Why is it needed?

In order to only build the necessary plugin when releasing a new version of Strapi.

### How to test it?

The build script should build the admin correctly and the dev mode inside the monorepo should work as it used to.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/admin/scripts/create-plugins-file.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const { join, resolve, relative } = require('path');
+const { camelCase } = require('lodash');
+const fs = require('fs-extra');
+
+const CORE_PLUGINS = ['content-type-builder', 'email', 'upload', 'i18n'];
```
Comment (alexandrebodin) [inline packages/core/admin/scripts/create-plugins-file.js:7]: i18n is in the plugins folder you can remove it here

File: packages/core/admin/scripts/build.js [hunk 2]
```diff
@@ -3,16 +3,28 @@
 const path = require('path');
 const webpack = require('webpack');
 const webpackConfig = require('../webpack.config');
+const {
+  getCorePluginsPath,
+  getPluginToInstallPath,
+  createPluginsFile,
+} = require('./create-plugins-file');
+
+const PLUGINS_TO_INSTALL = ['i18n'];
```
Comment (alexandrebodin) [inline packages/core/admin/scripts/build.js:12]: at the beginning of v4 we will keep users-permissions installed too so you should keep it here

File: packages/core/admin/package.json [hunk 3]
```diff
@@ -9,7 +9,9 @@
   "scripts": {
     "prepublishOnly": "node ./scripts/build.js",
     "test": "echo \"no tests yet\"",
-    "develop": "cross-env NODE_ENV=development webpack serve --config webpack.config.dev.js --progress profile",
+    "create:plugin-file": "node ./scripts/create-dev-plugins-file.js",
+    "dev": "cross-env NODE_ENV=development webpack serve --config webpack.config.dev.js --progress profile",
+    "develop": "yarn create:plugin-file && yarn dev",
     "develop:ce": "STRAPI_DISABLE_EE=true webpack serve --config webpack.config.dev.js --progress profile",
```
[the remark this answers is not in the document]
Comment (soupette) [reply packages/core/admin/package.json]: Marking this one resolved as the command is not needed in the admin anymore. In order to be in ce mode in the admin we just need to start the app using the `ce` command.

**Review by alexandrebodin — APPROVED** [review]
LGTM

**Pull request merged**